canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,147 @@
+package com.amazon.kinesis.video.canary.consumer;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.lang.Exception;
+import java.util.Date;
+import java.text.MessageFormat;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.ArrayList;
+
+import com.amazonaws.auth.SystemPropertiesCredentialsProvider;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsync;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatchAsyncClientBuilder;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideo;
+import com.amazonaws.services.kinesisvideo.AmazonKinesisVideoClientBuilder;
+import com.amazonaws.services.kinesisvideo.model.APIName;
+import com.amazonaws.services.kinesisvideo.model.GetDataEndpointRequest;
+import com.amazonaws.services.kinesisvideo.model.TimestampRange;
+import com.amazonaws.services.kinesisvideo.model.FragmentSelector;
+import com.amazonaws.services.cloudwatch.model.Dimension;
+import com.amazonaws.services.cloudwatch.model.MetricDatum;
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+public class WebrtcStorageCanaryConsumer {
+
+    private static void calculateFragmentContinuityMetric(CanaryFragmentList fragmentList, Date canaryStartTime, String streamName, String canaryLabel, SystemPropertiesCredentialsProvider credentialsProvider, String dataEndpoint, String region){
+        try{
+            TimestampRange timestampRange = new TimestampRange();
+            timestampRange.setStartTimestamp(canaryStartTime);
+            timestampRange.setEndTimestamp(new Date());
+
+            FragmentSelector fragmentSelector = new FragmentSelector();
+                fragmentSelector.setFragmentSelectorType("SERVER_TIMESTAMP");
+                fragmentSelector.setTimestampRange(timestampRange);
+
+            Boolean newFragmentReceived = false;
+
+            ExecutorService executorService = Executors.newFixedThreadPool(10);
+            Future<List<CanaryFragment>> listFragmentResult = executorService.submit(new CanaryListFragmentWorker(streamName, credentialsProvider, dataEndpoint, Regions.fromName(region), fragmentSelector));
```

Comment:
Does this really need an executor? Does it not return immediately? If it does lets remove the use of executor. If it does, please add a comment on why this is required. Also, the executor service needs to shut down. I do not see a clean up logic.

Reply:
Ah yes, the `callable` implementation I was referencing had an executor because it also used `runnable`, it shouldn't be needed here.